pkg/alertmanager/amcfg_test.go

```diff
@@ -39,6 +39,7 @@ func TestGenerateConfig(t *testing.T) {
 		baseConfig alertmanagerConfig
 		amConfigs  map[string]*monitoringv1alpha1.AlertmanagerConfig
 		expected   string
+		ctx        context.Context
```

Comment:
I think it would make more sense to have the context global to the test function. I don't think we need to test different contexts in the cases, it should remain the same for each of them.